File: node/test/api/jdy/workflow.test.js
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/08
*/

import { WorkflowApiClient } from '../../../src/api/jdy/workflow';
import { ApiKey, Host } from '../../../src/base/api_client';
import { appId, entryId } from './app.test';
import { dataId } from './form_data.test';

export const workflowTest = 'workflowTest';

describe('workflow api test', () => {
    const api = new WorkflowApiClient(ApiKey, Host);

    test('approvalComments', async () => {
        try {
            const response = await api.approvalComments(appId, entryId, dataId, {});
            expect(response.approveCommentList).toBeTruthy();
            console.log(response);
        } catch (err) {
            // 数据不存在
            console.log(err);
        }
    });
});

File: csharp/test/api/jdy/Workflow.Test.cs
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/14
*/
using Src.Api.Jdy;
using Src.Base;

namespace Test;

[Collection("Workflow")]
[TestCaseOrderer("XUnit.Project.Orderers.AlphabeticalOrderer", "Test")]
public class WorkflowTest
{

    private static WorkflowApiClient api = new WorkflowApiClient(Global.Host, Global.ApiKey);


    [Fact]
    public async void Test1()
    {
        var response = await api.approvalComments(AppTest.appId, AppTest.entryId, FormDataTest.dataId, new Dictionary<string, object>());
        Assert.Equal(4001, response.Value.GetProperty("code").GetInt64());
        Console.WriteLine(response);
    }
}